flaml/automl.py

```diff
@@ -507,7 +507,11 @@ def custom_metric(
                 True - retrain only after search finishes; False - no retraining;
                 'budget' - do best effort to retrain without violating the time
                 budget.
-            split_type: str, default="auto" | the data split type.
+            split_type: str or splitter object, default="auto" | the data split type.
+                Object must be a derived class of scikit-learn KFold
+                (https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html#sklearn.model_selection.KFold)
+                and have ``split`` and ``get_n_splits`` methods with the same signatures.
```

Comment:
Thanks. These all look good to me. 
One other thing we need to handle when calling split() in ml.py:  https://github.com/microsoft/FLAML/blob/dd60dbc5eb7e6d298c5f1506a8e0b7a8bb390052/flaml/ml.py#L319
When task == TS_FORCAST, always set shuffle=False to avoid data shuffling. 
The simplest way is to set this line 
https://github.com/microsoft/FLAML/blob/dd60dbc5eb7e6d298c5f1506a8e0b7a8bb390052/flaml/ml.py#L308
`shuffle=False if task == TS_FORECAST else True` and update https://github.com/microsoft/FLAML/blob/dd60dbc5eb7e6d298c5f1506a8e0b7a8bb390052/flaml/ml.py#L315 and https://github.com/microsoft/FLAML/blob/dd60dbc5eb7e6d298c5f1506a8e0b7a8bb390052/flaml/ml.py#L319 accordingly.

Reply:
If `shuffle = False if task == TS_FORECAST else True` at Line 308, not quite sure what needs to be updated at Line 315 and 319. What if they remain the same?

Reply:
please just remove line 319 after changing line 308, as 319 is already handled by 308.